worlds/messenger/transitions.py

```diff
@@ -0,0 +1,96 @@
+                break
+        else:
+            raise ValueError(f"Invalid target region for {plando_connection}")
+        region.entrances.remove(_entrance)
+
+    multiworld = world.multiworld
+    player = world.player
+    for plando_connection in plando_connections:
+        # get the connecting regions
+        reg1 = multiworld.get_region(plando_connection.entrance, player)
+        reg2 = multiworld.get_region(plando_connection.exit, player)
+
+        remove_dangling_exit(reg1)
+        remove_dangling_entrance(reg2)
+        # connect the regions
+        reg1.connect(reg2)
+
+        # pretend the user set the direction as "both" regardless of what they actually put on coupled
+        if ((world.options.shuffle_transitions == ShuffleTransitions.option_coupled
+             or plando_connection.direction == "both")
+                and plando_connection.exit in RANDOMIZED_CONNECTIONS):
+            remove_dangling_exit(reg2)
+            remove_dangling_entrance(reg1)
+            reg2.connect(reg1)
+
+
+def shuffle_transitions(world: "MessengerWorld") -> None:
+    multiworld = world.multiworld
+    player = world.player
+    coupled = world.options.shuffle_transitions == ShuffleTransitions.option_coupled
+
+    def disconnect_entrance() -> None:
+        child_region.entrances.remove(entrance)
+        entrance.connected_region = None
+
+        er_type = EntranceType.ONE_WAY if child == "Glacial Peak - Left" else \
+            EntranceType.TWO_WAY if child in RANDOMIZED_CONNECTIONS else EntranceType.ONE_WAY
+        if er_type == EntranceType.TWO_WAY:
+            mock_entrance = parent_region.create_er_target(entrance.name)
+        else:
+            mock_entrance = child_region.create_er_target(child)
+
+        entrance.randomization_type = er_type
+        mock_entrance.randomization_type = er_type
+
+    for parent, child in RANDOMIZED_CONNECTIONS.items():
+        if child == "Corrupted Future":
+            entrance = multiworld.get_entrance("Artificer's Portal", player)
+        elif child == "Tower of Time - Left":
+            entrance = multiworld.get_entrance("Artificer's Challenge", player)
+        else:
+            entrance = multiworld.get_entrance(f"{parent} -> {child}", player)
+        parent_region = entrance.parent_region
+        child_region = entrance.connected_region
+        entrance.world = world
+        disconnect_entrance()
+
+    plando = world.options.plando_connections
+    if plando:
+        connect_plando(world, plando)
+
+    result = randomize_entrances(world, coupled, {0: [0]})
+
+    world.transitions = sorted(result.placements, key=lambda entrance: TRANSITIONS.index(entrance.parent_region.name))
+
+    for transition in world.transitions:
+        if "->" not in transition.name:
+            continue
+        transition.parent_region.exits.remove(transition)
+        transition.name = f"{transition.parent_region} -> {transition.connected_region}"
```

Comment:
got curious and tracked down where the player names were coming from and it's this line
but just simply changing to 
```suggestion
        transition.name = f"{transition.parent_region.name} -> {transition.connected_region.name}"
```
runs into some entrance cache issues (that i'm sure you're more apt to handle than i)

Reply:
I'm curious what you mean by entrance cache issues. The only place I can think of this might exist would be the multiworld's CollectionState. If that state specifically does any sweeps before this block that could cause issues, but since i'm changing my personal caches manually, those should all be fine, and I don't store any of the all states I use.

Reply:
i got errors about adding entrances with the same name when i just tried to do that simple change, didn't write them down so i'd have to reproduce

Reply:
Ah ok managed to repro it. What happened is there's a few transitions where you can go from one to the other with nothing notable between (i.e. Quillshroom - Bottom Left -> Quillshroom - Bottom Right is a connection that always exists) and it connected the edges to each other.